src/Parser/Core/Modules/DistanceMoved.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+const debug = false;
+
+class DistanceMoved extends Module {
+    lastPositionUpdate = null;
+    totalDistanceMoved = 0;
+
+    // Events
+    on_cast(event) {
+        this.updatePlayerPosition(event);
+    }
+
+    on_damage(event) {
+        this.updatePlayerPosition(event);
+    }
+
+    on_energize(event) {
+        this.updatePlayerPosition(event);
+    }
+
+    on_heal(event) {
+        this.updatePlayerPosition(event);
+    }
+
+    on_absorbed(event){
+        this.updatePlayerPosition(event);
+    }
+
+    // Data parsing
+    calculateDistance(x1, y1, x2, y2) {
+        return Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)) / 100;
+    }
+
+    updateTotalDistance(event) {
+        if (!this.lastPositionUpdate) {
+            return;
+        }
+        this.totalDistanceMoved += this.calculateDistance(this.lastPositionUpdate.x, this.lastPositionUpdate.y, event.x, event.y);
+    }
+
+    updatePlayerPosition(event) {
+        if (!event.x || !event.y) {
+            return;
+        }
+        this.updateTotalDistance(event);
+        this.lastPositionUpdate = event;
+    }
+
+    statistic() {
+        debug && console.log(`Total distance moved: ${this.totalDistanceMoved} yds`);
+        return (
+            <StatisticBox
+            icon={<SpellIcon id={SPELLS.FRENETIC_SPEED_TALENT.id} />}
+            value={`${formatNumber(this.totalDistanceMoved)} yds moved`}
```

Comment:
How about making this show the total amount, e.g. `21,342 yards` (or w/e)? The label already shows what it is.

How much confidence do you have in the accuracy of this? I'm half thinking it might warrant adding a `≈` in front of the distance.

Reply:
Yeah it probably needs that in front of it, should I still format the number or not?

Reply:
Yeah I think the extra detail would be great.

Reply:
I'll add it in front.